funnel/models/project.py

```diff
@@ -879,6 +879,14 @@ def roles_for(self, actor=None, anchors=()):
             if crew_membership is not None:
                 roles.update(crew_membership.offered_roles())
 
+            if (
+                self.rsvp_for(actor) is not None
+                or self.participants.filter_by(user=actor).one_or_none() is not None
```

Comment:
I believe you can do `actor in self.participants`

Reply:
This may be simpler using the `grants` mechanism. Make sure the `rsvp` relationship is defined first as it's more commonly used.